Game/LambdaHack/Server/Commandline.hs

```diff
@@ -9,162 +12,277 @@ import Game.LambdaHack.Common.Prelude
+        , ssavePrefixCli    = savePrefix
+        , sfrontendTeletype = frontendTeletype
+        , sfrontendNull     = frontendNull
+        , sfrontendLazy     = frontendLazy
+        , sdbgMsgCli        = dbgMsgCli
+        , sstopAfterSeconds = stopAfterSeconds
+        , sstopAfterFrames  = stopAfterFrames
+        }
+    }
+ where
+   serToChallenge :: Maybe Int -> (Bool, Challenge)
+   serToChallenge Nothing  = (False, defaultChallenge)
+   serToChallenge (Just n) = (True, Challenge n False False)
+
+knowMapP :: Parser Bool
+knowMapP =
+  switch (  long "knowMap"
+         <> help "reveal map for all clients in the next game" )
+
+knowEventsP :: Parser Bool
+knowEventsP =
+  switch (  long "knowEvents"
+         <> help "show all events in the next game (needs --knowMap)" )
+
+knowItemsP :: Parser Bool
+knowItemsP =
+  switch (  long "knowItems"
+         <> help "auto-identify all items in the next game (needs --knowEvents)" )
+
+sniffInP :: Parser Bool
+sniffInP =
+  switch (  long "sniffIn"
+         <> help "display all incoming commands on console" )
+
+sniffOutP :: Parser Bool
+sniffOutP =
+  switch (  long "sniffOut"
+         <> help "display all outgoing commands on console" )
+
+allClearP :: Parser Bool
+allClearP =
+  switch (  long "allClear"
+         <> help "let all map tiles be translucent" )
+
+boostRandItemP :: Parser Bool
+boostRandItemP =
+  switch (  long "boostRandomItem"
+         <> help "pick a random item and make it very common" )
+
+gameModeP :: Parser (Maybe (GroupName ModeKind))
+gameModeP = optional $ toGameMode <$>
+  strOption (  long "gameMode"
+            <> metavar "MODE"
+            <> help "start next game in the given mode" )
+ where
+  toGameMode :: String -> GroupName ModeKind
+  toGameMode s  = toGroupName (T.pack s)
+
+automateAllP :: Parser Bool
+automateAllP =
+  switch (  long "automateAll"
+         <> help "give control of all UI teams to computer" )
+
+keepAutomatedP :: Parser Bool
+keepAutomatedP =
+  switch (  long "keepAutomated"
+         <> help "keep factions automated after game over" )
+
+newGameP :: Parser (Maybe Int)
+newGameP =
+  optional $ option auto
+    ( long "newGame"
+   <> help "start a new game, overwriting the save file, with difficulty for all UI players set to this value"
+   <> metavar "INT" )
+
+stopAfterSecsP :: Parser (Maybe Int)
+stopAfterSecsP =
+  optional $ option auto
+    ( long "stopAfterSeconds"
+   <> help "exit game session after around n seconds"
+   <> metavar "INT" )
+
+stopAfterFramesP :: Parser (Maybe Int)
+stopAfterFramesP =
+  optional $ option auto
+    ( long "stopAfterFrames"
+   <> help "exit game session after around n frames"
+   <> metavar "INT" )
+
+benchmarkP :: Parser Bool
+benchmarkP =
+  switch (  long "benchmark"
+         <> help "restrict file IO, print stats" )
+
+setDungeonRngP :: Parser (Maybe R.StdGen)
+setDungeonRngP = optional $
+  option auto (  long "setDungeonRng"
+              <> metavar "DUNGEON_RNG"
```

Comment:
I'd call it just RNG_SEED or just SEED, because it's exactly the same type and role as in the next option.